src/mscorlib/src/System/Text/Encoding.cs

```diff
@@ -884,6 +884,19 @@ public virtual int GetByteCount(String s)
         [Pure]
         public abstract int GetByteCount(char[] chars, int index, int count);
 
+        // Returns the number of bytes required to encode a string range.
+        //
+        [Pure]
+        public virtual int GetByteCount(string s, int index, int count)
```

Comment:
The API proposals can have bugs too, and the fact that they got approved does not make them bug-free...